### PR TITLE
[IMP] mail: small activity cleanup

### PR DESCRIPTION
**Pull request by phenix-factory**
Use the activity persona information instead of the custom code and user_id array.

PR enterprise: https://github.com/odoo/enterprise/pull/75404

### CODE REVIEW

File: addons/mail/static/src/core/web/activity_list_popover_item.xml [hunk 1]
```diff
@@ -29,10 +29,10 @@
                 </button>
             </div>
             <div class="d-flex align-items-center flex-wrap mx-3">
-                <img t-if="props.activity.user_id[0]" class="me-2 rounded o_object_fit_cover" t-att-src="activityAssigneeAvatar" style="max-width: 1.5rem; max-height: 1.5rem;"/>
+                <img t-if="props.activity.persona.userId" class="me-2 rounded o_object_fit_cover" t-att-src="props.activity.persona.avatarUrl" style="max-width: 1.5rem; max-height: 1.5rem;"/>
                 <div class="mt-1">
-                    <t t-if="props.activity.user_id[0]">
-                        <small class="text-truncate" t-esc="props.activity.user_id[1]"/>
+                    <t t-if="props.activity.persona.userId">
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/activity_list_popover_item.xml:34]: I don't think we need to keep a check on `userId` in this template if we no longer use `userId`.
Maybe a check on `persona`, but I'd argue we should always send it in data with activity.

**Review by seb-odoo — APPROVED** [review]
Apart from small comments, the idea behind the cleanup looks fine.

File: addons/mail/static/src/core/web/activity_list_popover_item.xml (hunk 1 above)
Comment (phenix-factory) [reply addons/mail/static/src/core/web/activity_list_popover_item.xml:34]: If there is always a persona with the activity, I guess it's not useful to check it at all

File: addons/mail/static/src/core/web/activity_list_popover_item.xml [hunk 2]
```diff
@@ -29,12 +29,10 @@
...
             <div class="d-flex align-items-center flex-wrap mx-3">
-                <img t-if="props.activity.user_id[0]" class="me-2 rounded o_object_fit_cover" t-att-src="activityAssigneeAvatar" style="max-width: 1.5rem; max-height: 1.5rem;"/>
+                <img class="me-2 rounded o_object_fit_cover" t-att-src="props.activity.persona.avatarUrl" style="max-width: 1.5rem; max-height: 1.5rem;"/>
                 <div class="mt-1">
-                    <t t-if="props.activity.user_id[0]">
-                        <small class="text-truncate" t-esc="props.activity.user_id[1]"/>
-                        <small t-if="props.activity.state !== 'today'" class="mx-1">-</small>
-                    </t>
+                  <small class="text-truncate" t-esc="props.activity.persona.name"/>
+                  <small t-if="props.activity.state !== 'today'" class="mx-1">-</small>
```
Comment (seb-odoo) [inline addons/mail/static/src/core/web/activity_list_popover_item.xml:35]: wrong indent

**Review by seb-odoo — APPROVED** [review]
@robodoo delegate+

**Comment by phenix-factory** [comment]
@robodoo r+

**Pull request closed without merging**